File: src/main/java/de/hse/model/Product.java
package de.hse.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name = "product_id")
    private long productId;
    private String name;
    private String description;
    private double price;
    private String base;
}

File: src/web/src/app/model/product.ts
export class Product{
  productId?: number;
  name?: string;
  description?: string;
  price?: number;
  base?: string;

  constructor(
    productId?: number,
    name?: string,
    description?: string,
    price?: number,
    base?: string
  ) {
      this.productId = productId,
      this.name = name,
      this.description = description,
      this.price = price,
      this.base = base
    }
}
